c_src/Makefile

```diff
@@ -14,13 +14,13 @@ C_SRC_DIR = $(CURDIR)
 C_SRC_OUTPUT ?= $(CURDIR)/../priv/$(PROJECT).so
 
 # System type and C compiler/flags.
-
+MACHINE_SYS := $(shell uname -m)
 UNAME_SYS := $(shell uname -s)
 ifeq ($(UNAME_SYS), Darwin)
 	CC ?= cc
-	CFLAGS ?= -O3 -std=c99 -arch x86_64 -finline-functions -Wall -Wmissing-prototypes
-	CXXFLAGS ?= -O3 -arch x86_64 -finline-functions -Wall
-	LDFLAGS = -arch x86_64 -flat_namespace -undefined suppress -bundle
+	CFLAGS ?= -O3 -std=c99 -arch $(MACHINE_SYS) -finline-functions -Wall -Wmissing-prototypes
```

Comment:
Can you please check and see that it does not break x86_64. I tried on my mac and ubuntu `uname -m` returns x86_64 - so I think you are ok.

Reply:
Yes; murali said that he will test in a bit on his x86. It's reassuring that `uname -m` seems to give expected behavior though - thanks for checking.